src/v/storage/disk_log_impl.cc

```diff
@@ -1026,9 +1036,25 @@ ss::future<> disk_log_impl::housekeeping(housekeeping_config cfg) {
      * there is a need to run it separately.
      */
     if (config().is_compacted() && !_segs.empty()) {
+        // TODO: unify error handling.
         if (config::shard_local_cfg().log_compaction_use_sliding_window()) {
-            auto compacted = co_await sliding_window_compact(
-              cfg.compact, new_start_offset);
+            cfg.compact.asrc = &_compaction_as;
+            auto did_compact_fut = co_await ss::coroutine::as_future(
+              sliding_window_compact(cfg.compact, new_start_offset));
+            if (did_compact_fut.failed()) {
+                auto eptr = did_compact_fut.get_exception();
+                if (!ssx::is_shutdown_exception(eptr)) {
+                    std::rethrow_exception(eptr);
+                }
+            }
+            if (_compaction_as.abort_requested()) {
+                vlog(
+                  gclog.debug,
+                  "Compaction of {} stopped because of shutdown",
+                  config().ntp());
+                co_return;
+            }
+            bool compacted = did_compact_fut.get();
```

Comment:
if `did_compact_fut` failed, but it wasn't a shutdown exception and an abort wasn't requested, then it looks like we'd fall through to here and this would panic when we called `get` on the failed future?

Reply:
Fair point. They mostly should be caught by the check on the abort source above, but you're right that there's possible room for error if we don't check the exception in that if block too. Will update

Reply:
Done